src/auth/digest/eDirectory/ldap_backend.cc

```diff
@@ -203,8 +203,8 @@ getpassword(char *login, char *realm)
     char **value = NULL;
     char *password = NULL;
     int retry = 0;
-    char filter[8192];
-    char searchbase[8192];
+    char filter[8192] = {};
+    char searchbase[8192] = {};
```

Comment:
Technically, this is a performance regression because we are now zeroing 16KB worth of stack on every getpassword() call. However, I assume that this function is called only at helper startup or we just do not care enough about sample helper performance to worry about such minor things.

Reply:
FYI this function is called on every lookup

Reply:
Does this initialization trick resolve the issue?
```suggestion
    char searchbase[8192];
    *searchbase = 0;
```

Reply:
The answer depends on what you mean by "issue":

* If you are asking about the performance regression, then, yes, zeroing just the first character would address that concern.

* If you are asking about the initialization problem being fixed by this PR, then I do not know the answer -- it depends on how these buffers are used, and I have not checked that. I bet @kinkie knows the answer.

@kinkie, please do not forget that there are _two_ buffers in this PR scope so if we decide to initialize just the first character it has to be done for both of them despite the above suggestions focusing on just one of the buffers.

Needless to say, there is a way to avoid all these low-level initialization tricks and risks, but that would require code refactoring that is probably beyond the current scope of this PR.

Reply:
Q: from a performance point of view, if our only aim is to ensure the first char is \0, which is more efficient among:

1. char s[1024] = {};
2. char string[1024] = { 0 };
3. char string[1024] = "";
4. char string[1024]; string[0] = 0;

?

Reply:
The last item (number four) is faster from C++11 rules point of view. That item initializes nothing and then sets the first array item. AFAICT, that is the absolute minimum of the (required by the question) initialization work for a stack-allocated array.

AFAICT, the first three items are identical in terms of what a non-optimizing compiler should produce per C++11 rules -- they all initialize the _whole_ `string` array with 1024 zeros. The cppreference site does not have all these rules in one place AFAICT, but you can find the appropriate rules in various places talking about initialization: Items one and three are covered at https://en.cppreference.com/w/cpp/language/zero_initialization and item two (after the type conversions) is covered at https://en.cppreference.com/w/cpp/language/aggregate_initialization#Character_arrays

FWIW, you can see whole-array memset()-calling initialization for your first three items in my basic [tests](https://godbolt.org/z/1774nqceG).

If an optimizing compiler knows that it does not need to initialize the whole thing, then, I am _guessing_, it does not have to do that.